src/devices/src/virtio/net/persist.rs

```diff
@@ -23,10 +24,39 @@ use super::{NUM_QUEUES, QUEUE_SIZE};
 use crate::virtio::persist::{Error as VirtioStateError, VirtioDeviceState};
 use crate::virtio::{DeviceState, TYPE_NET};
 
-#[derive(Clone, Versionize)]
+#[derive(Debug, Default, Clone, Versionize)]
 // NOTICE: Any changes to this structure require a snapshot version bump.
 pub struct NetConfigSpaceState {
-    guest_mac: [u8; MAC_ADDR_LEN],
+    #[version(end = 2, default_fn = "def_net_config_old")]
+    guest_mac_old: [u8; utils::net::mac::MAC_ADDR_LEN],
+    #[version(start = 2, de_fn = "de_net_config", ser_fn = "ser_net_config")]
+    guest_mac: Option<MacAddr>,
```

Comment:
Why not `Option<[u8; utils::net::mac::MAC_ADDR_LEN]>`, I am asking because this way it would not be necessary to make the utils crate dependent on Versionize

Reply:
We have `MacAddr` as type that represents MAC address and basically just a wrapper around `[u8; MAC_ADDR_LEN]`. 
I think it is better to use it, then just an array of bytes. It makes code easier to understand, without a bunch of `MacAddr::from_bytes_unchecked` calls.
(On this note I probably should make `guest_mac_old` also be `MacAddr`)